src/AWS.Deploy.Orchestration/CdkProjectHandler.cs

```diff
@@ -53,23 +48,34 @@ public async Task CreateCdkDeployment(OrchestratorSession session, CloudApplicat
             {
                 // Create a new temporary CDK project for a new deployment
                 _interactiveService.LogMessageLine($"Generating a {recommendation.Recipe.Name} CDK Project");
-                cdkProjectPath = CreateCdkProjectForDeployment(recommendation, session);
+                cdkProjectPath = CreateCdkProject(recommendation, session);
             }
-            
+
             // Write required configuration in appsettings.json
             var appSettingsBody = _appSettingsBuilder.Build(cloudApplication, recommendation, session);
             var appSettingsFilePath = Path.Combine(cdkProjectPath, "appsettings.json");
-            using (var appSettingsFile = new StreamWriter(appSettingsFilePath))
+            await using var appSettingsFile = new StreamWriter(appSettingsFilePath);
+            await appSettingsFile.WriteAsync(appSettingsBody);
+
+            return cdkProjectPath;
+        }
+
+        public async Task DeployCdkProject(OrchestratorSession session, string cdkProjectPath, Recommendation recommendation)
+        {
+            var recipeInfo = $"{recommendation.Recipe.Id}_{recommendation.Recipe.Version}";
+            var environmentVariables = new Dictionary<string, string>
             {
-                await appSettingsFile.WriteAsync(appSettingsBody);
-            }
+                { EnvironmentVariableKeys.AWS_EXECUTION_ENV, recipeInfo }
```

Comment:
We are messing up the execution env which is configured `App.cs`. It should always start with the current value set for execution env which in the CLI would be `aws-dotnet-deploy-cli` and then we can add recipe info as additional information.

Reply:
This is already taken care [here](https://github.com/aws/aws-dotnet-deploy/blob/4c541282dce9a3582afcfe1621b99b6253947e45/src/AWS.Deploy.CLI/Utilities/CommandLineWrapper.cs#L127)